astropy/io/ascii/tests/test_c_reader.py

```diff
@@ -12,6 +12,7 @@
 import numpy as np
 import pytest
 from numpy import ma
+from numpy.testing import assert_allclose, assert_equal
```

Comment:
This looks half-finished. As you saw, we can just inline `assert_equal` instead of replacing it with `numpy.testing.assert_equal`. A couple call sites seem to remain, let's smoke them out !
```suggestion
from numpy.testing import assert_allclose
```

Reply:
> This looks half-finished. As you saw, we can just inline `assert_equal` instead of replacing it with `numpy.testing.assert_equal`. A couple call sites seem to remain, let's smoke them out !

@neutrinoceros there has been an ongoing discussion from
https://github.com/astropy/astropy/pull/15839#issuecomment-1885134788, https://github.com/astropy/astropy/pull/15839#issuecomment-1892419145 and following about when to better use simple `assert` vs. Numpy's `assert_equal` on array-like operands. I am afraid there is no uncontroversial golden path where to use what, but my last set of comments and suggestion was trying to find a reasonable balance of using both options. Do you think `assert x == y` should be always preferred even on numerical ndarrays?

Reply:
Oh, sorry I missed it !

> Do you think assert x == y should be always preferred even on numerical ndarrays?

no ! I was trying to be strictly conservative in my review, but I agree that this pattern is best avoided for ndarrays !